pkg/userdata/generate/generate.go

```diff
@@ -313,34 +331,31 @@ func NewInput(clustername string, masterIPs []string, kubernetesVersion string)
 	if err != nil {
 		return nil, err
 	}
-	caPemBlock, _ := pem.Decode(osCert.CrtPEM)
-	if caPemBlock == nil {
-		return nil, errors.New("failed to decode ca cert pem")
-	}
-	caCrt, err := stdlibx509.ParseCertificate(caPemBlock.Bytes)
+	adminCrt, err := x509.NewCertificateFromCSR(caCrt, caKey, ccsr, opts...)
 	if err != nil {
 		return nil, err
 	}
-	caKeyPemBlock, _ := pem.Decode(osCert.KeyPEM)
-	if caKeyPemBlock == nil {
-		return nil, errors.New("failed to decode ca key pem")
-	}
-	caKey, err := stdlibx509.ParseECPrivateKey(caKeyPemBlock.Bytes)
-	if err != nil {
-		return nil, err
+
+	// Generate Etcd CA.
+	opts = []x509.Option{
+		x509.RSA(false),
+		x509.Organization("talos-etcd"),
+		x509.NotAfter(time.Now().Add(87600 * time.Hour)),
 	}
-	adminCrt, err := x509.NewCertificateFromCSR(caCrt, caKey, ccsr, opts...)
+	etcdCert, err := x509.NewSelfSignedCertificateAuthority(opts...)
 	if err != nil {
 		return nil, err
 	}
 
 	certs := &Certs{
 		AdminCert: base64.StdEncoding.EncodeToString(adminCrt.X509CertificatePEM),
 		AdminKey:  base64.StdEncoding.EncodeToString(adminKey.KeyPEM),
-		OsCert:    base64.StdEncoding.EncodeToString(osCert.CrtPEM),
-		OsKey:     base64.StdEncoding.EncodeToString(osCert.KeyPEM),
+		EtcdCert:  base64.StdEncoding.EncodeToString(etcdCert.CrtPEM),
```

Comment:
Same as above:  EtcdPeerCert, EtcdServiceCert

Reply:
the above is just internal, I'm not sure it matters much ( ex, we don't really distinguish between the OS level certs ( osd )).